main.go

```diff
@@ -115,8 +137,60 @@ func main() {
 	}
 
 	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+	if err := mgr.Start(ctx); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
+
+// Tee consumes the received channel and mirrors the messages into 2 new channels.
+func Tee[T any](ctx context.Context, in <-chan T) (_, _ <-chan T) {
```

Comment:
This fanout helper doesn't need to be public:

```suggestion
func tee[T any](ctx context.Context, in <-chan T) (_, _ <-chan T) {
```

Reply:
Agree. I think it can't be used anyways since we'll end up with a dependency cycle